solr/core/src/java/org/apache/solr/servlet/CoordinatorHttpSolrCall.java

```diff
@@ -92,8 +93,9 @@ public static SolrCore getCore(
                 "synthetic collection: {} does not exist, creating.. ", syntheticCollectionName);
           }
           createColl(syntheticCollectionName, solrCall.cores, confName);
+          // Try to get core again now that we have created the synthetic core
+          core = solrCall.getCoreByCollection(collectionName, isPreferLeader);
         }
-        SolrCore core = solrCall.getCoreByCollection(syntheticCollectionName, isPreferLeader);
```

Comment:
This call here, was ending up in an endless loop and stackoverflow expection with multiple coordinator nodes. This is because we don't createColl above on the second node since the first already created the synthetic collection and we end up just looping. To fix this I moved this into the block after creating the collection.

Reply:
If first has created a `SyhtheticCall` then other should get zk notification for that?

Reply:
The collection is created with a single replica on the node that initiated the create collection command. So say there are two coordinator nodes: coordinator-1 and coordinator-2.

coordinator-1 gets a call and creates the synthetic collection with one replica that is on coordinator-1
coordinator-2 then gets a call and it sees the collection exists so it adds a replica on itself